Assets/MixedRealityToolkit/Inspectors/Utilities/MixedRealitySearchUtility.cs

```diff
@@ -0,0 +1,307 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor.Search
+{
+    /// <summary>
+    /// Utility for searching profile 
+    /// </summary>
+    public static partial class MixedRealitySearchUtility
+    {
+        /// <summary>
+        /// Field names that shouldn't be displayed in a profile field search.
+        /// </summary>
+        private static readonly HashSet<string> serializedPropertiesToIgnore = new HashSet<string>()
+        {
+            // Unity base class fields
+            "m_Name",
+            "m_Script",
+            "m_Enabled",
+            "m_GameObject",
+            "m_ObjectHideFlags",
+            "m_CorrespondingSourceObject",
+            "m_PrefabInstance",
+            "m_PrefabAsset",
+            "m_EditorHideFlags",
+            "m_EditorClassIdentifier",
+            // Profile base class fields
+            "isCustomProfile",
+        };
+
+        /// <summary>
+        /// Field types that don't need their child properties displayed.
+        /// </summary>
+        private static readonly HashSet<string> serializedPropertyTypesToFlatten = new HashSet<string>()
```

Comment:
If we switch to a simpler, render the profile if a match was found. This should be removed I believe?

That approach should also be faster since we can bail early if any match was found and just proceed to rendering